### PR TITLE
List RRD directory once per monitor poll

### PR DESCRIPTION
**Pull request by psafont**
Before it was being listed 3 times per poll, once per each RRD monitored.

### CODE REVIEW

File: ocaml/xapi/monitor_types.ml [hunk 1]
```diff
@@ -37,7 +37,6 @@ let vif_device_of_string x =
 let find_rrd_files prefix =
```
Comment (robhoes) [inline ocaml/xapi/monitor_types.ml]: Should this now be `let find_rrd_files ()`?

Comment (psafont) [reply ocaml/xapi/monitor_types.ml]: indeed!

File: ocaml/xapi/monitor_mem_vms.ml [hunk 2]
```diff
@@ -55,7 +55,9 @@ let set_changes ?except () =
       Mcache.transfer_map ?except ~source:Mcache.vm_memory_tmp ~target:Mcache.vm_memory_cached
     )
 
-let update () =
+let update rrd_files =
+  let is_vm_rrd = (Astring.String.is_prefix ~affix:Xapi_globs.metrics_prefix_mem_vms) in
```
Comment (lindig) [inline ocaml/xapi/monitor_mem_vms.ml]: No '(' .. ')' needed.

File: ocaml/xapi/monitor_pvs_proxy.ml [hunk 3]
```diff
@@ -75,7 +75,9 @@ let pvs_proxy_status_of_int = function
   | 4 -> `incompatible_protocol_version
   | _ -> failwith "Unknown status"
 
-let update () =
+let update rrd_files =
+  let is_proxy_rrd = (Astring.String.is_prefix ~affix:Xapi_globs.metrics_prefix_pvs_proxy) in
```
Comment (lindig) [inline ocaml/xapi/monitor_pvs_proxy.ml]: No '(' .. ')' needed.

**Pull request merged**